### ISSUE TITLE
futimens system call does not work correctly on Windows filesystems

### ISSUE THREAD

**Issue by pcc** [Microsoft/BashOnWindows#1399]
* A brief description

1. Create a file `futimens.c` with the following contents:
```
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>

int main() {
  int fd = open("foo", O_RDWR | O_CREAT, 0644);
  write(fd, "foo", 3);
  struct timespec times[2] = {};
  futimens(fd, times);
  close(fd);
}
```
2. `make futimens`
3. `./futimens` from a Windows filesystem
4. `TZ=C stat foo | grep Modify:`

* Expected results
`Modify: 1970-01-01 00:00:00.000000000 +0000`

* Actual results (with terminal output if applicable)
`Modify: 2016-11-21 02:17:51.412669400 +0000`

* Your Windows build number
14393

* Steps / All commands required to reproduce the error from a brand new installation 
1. `sudo apt-get install gcc make`
2. see above

* Strace of the failing command
```
[...]
open("foo", O_RDWR|O_CREAT, 0644)       = 3
write(3, "foo", 3)                      = 3
utimensat(3, NULL, {{0, 0}, {0, 0}}, 0) = 0
close(3)                                = 0
```

* Required packages and commands to install
see above

**Comment by benhillis** [comment]
I just verified that this is working as expected in a recent insider build. I was running 14975 but I suspect this has been fixed for quite some time. Marking as fixed.

**Comment by pcc** [comment]
Can you confirm that this is the behaviour on DrvFs (i.e. `/mnt/X`)?

**Comment by benhillis** [comment]
You're right, the file is still showing a non-zero modified time on DrvFs.

**Comment by benhillis** [comment]
Tried this on 1803 and it is working as expected.
